### PR TITLE
feat: added healthcheck to docker

### PR DESCRIPTION
**Pull request by sum35h**
## Description
Added **healthcheck** to the fat dockerfile. The `healthcheck.sh` script used `supervisorctl` to check for the status of all the process to determine the health.
Added healthcheck to thin server and editor docker file.

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Manually

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feat/docker-healthcheck 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.47 **(0.01)** | 37.92 **(0.01)** | 36.15 **(0)** | 56.69 **(0.01)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0.78)** | 70.59 **(0)** | 100 **(0)** | 93.33 **(0.95)**</details>

### CODE REVIEW

File: app/client/Dockerfile [hunk 1]
```diff
@@ -16,4 +16,5 @@ COPY ./docker/templates/nginx-app-https.conf.template /nginx-app-https.conf.temp
 
 # This is the script that is used to start Nginx when the Docker container starts
 COPY ./docker/start-nginx.sh /start-nginx.sh
+HEALTHCHECK --interval=15s --timeout=15s --start-period=15s --retries=3 CMD curl -f http://localhost:80/ || exit 1
```
Comment (sharat87) [inline app/client/Dockerfile:19]: What's the reason for `|| exit 1`? If the curl command failed, it would anyway exit with a non-zero status code right?

**Review by sharat87 — APPROVED** [review]

**Comment by renatocron** [comment]
Need to add some way to check if the HTTPS was already configured, or tho test the editor directly
After setting up the letscencrypt/https, I changed the healthcheck script to use port 443:

``` 
if [[ $(curl -k -s -w "%{http_code}\n" https://localhost:443/ -o /dev/null) -ne 200 ]]; then
   echo 'ERROR: Editor is down';
   healthy=false
fi
```

**Comment by sharat87** [comment]
@renatocron, the latest version of that check actually does take https into account, by adding the `-L` argument to `curl`. With this, `curl` will just accept the redirect response from the http endpoint and test the https endpoint as well. See https://github.com/appsmithorg/appsmith/pull/13947/files#diff-359045826464e68364521558edab24d3a9b3de87bdbeeec9cedb1869e0560586.

Also, in general, the `:443` is not needed when using `https://` in the URL. This is because 443 is the default port of https URLs.

**Comment by renatocron** [comment]
@sharat87 thank you, I should have check the file was modified 5 days ago before commenting here!

The https://github.com/appsmithorg/appsmith/blob/release/deploy/docker/scripts/healthcheck.sh indeed does work!

**Pull request merged**